File: Java/Exercism/PerfectNumbers.java
class NaturalNumber {

    public String outcome = "";
    
    NaturalNumber(int number) {
        int Result = 0;

        if (number <= 0){
            throw new IllegalArgumentException("You must supply a natural number (positive integer)");
        }
        
        for (int i = 1; i < number; i++){
            if (number % i == 0){
                Result += i;
            }
        } 
        if (Result > number){
            outcome = "abundant";
        }
        if (Result < number){
            outcome = "deficient";
        }
        if (Result == number){
            outcome = "perfect";
        }
    }
    
    Classification getClassification() {
        
        if (outcome == "abundant"){
            return Classification.ABUNDANT;
        }
        else if (outcome == "perfect"){
            return Classification.PERFECT;
        }
        else
            return Classification.DEFICIENT;
        }
}

File: JavaScript/Exercism/PerfectNumbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const classify = number => {

    let answer = 0;
  
    if (number < 1){
      throw new Error('Classification is only possible for natural numbers.');
    }
    
    for (let i = 1; i < number; i++){ 
      if (number % i == 0){
        answer += i;
      }
    }
  
    if (answer > number){
      return "abundant";
    }
    else if (answer == number){
      return "perfect";
    }
    else {
      return "deficient";
    }
    
    // split it in half
    // start i at 1
    // don't halve 2
    // for value in range(1, half_value: value + value
    // output string answer
  
  };
  